# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/wr/WebRenderBridgeChild.h
# Commit: 9bfd4e60ec4e
# Full Hash: 9bfd4e60ec4e36b680c1722a3bd3e4b7359f7e36
# Author: Doug Thayer <dothayer@mozilla.com>
# Date: 2019-03-23 09:46:24
# Regressor Bug: 1441308
# File Overlap Count: 2
# Description:
#   Bug 1441308 - Always send parent commands when sending mDestroyedActors r=kats,sotaro
#   
#   If we try to send them separately as we were before, we can run into
#   cases where we try to destroy the actors and then send the OpRemoveTexture,
#   which crashes.
# ==============================================================================

diff -r 8761ce294d2c -r 9bfd4e60ec4e gfx/layers/wr/WebRenderBridgeChild.h
--- a/gfx/layers/wr/WebRenderBridgeChild.h	Fri Mar 22 18:29:00 2019 +0000
+++ b/gfx/layers/wr/WebRenderBridgeChild.h	Fri Mar 22 18:29:04 2019 +0000
@@ -66,6 +66,9 @@
 
   void AddWebRenderParentCommand(const WebRenderParentCommand& aCmd,
                                  wr::RenderRoot aRenderRoot);
+  bool HasWebRenderParentCommands(wr::RenderRoot aRenderRoot) {
+    return !mParentCommands[aRenderRoot].IsEmpty();
+  }
 
   void UpdateResources(wr::IpcResourceUpdateQueue& aResources,
                        wr::RenderRoot aRenderRoot);